Components/Motors/VESC.cs

```diff
@@ -0,0 +1,190 @@
+        {
+            float Output = this.Filter.GetOutput();
+            while (!this.Filter.IsSteadyState())
+            {
+                if (Stopped) { SetSpeedDirectly(0); }
+                else
+                {
+                    this.Filter.Feed(this.TargetSpeed);
+                    SetSpeedDirectly(this.Filter.GetOutput());
+                }
+                Thread.Sleep(Constants.DEFAULT_MIN_THREAD_SLEEP);
+            }
+            OngoingSpeedThread = false;
+        }
+
+        /// <summary> Creates a new thread for setting speed during motor filtering output </summary>
+        /// <returns> A new thread for changing the motor speed. </returns>
+        private Thread SetSpeedThreadFactory() { return new Thread(new ThreadStart(SetSpeedThread)); }
+
+        /// <summary>
+        /// Sets the motor speed. Output may vary from the given value under the following conditions:
+        /// - Input exceeds maximum speed. Capped to given maximum.
+        /// - Filter changes value. Filter's output used instead.
+        ///     (If filter is null, this does not occur)
+        /// - The motor is disabled. You must first re-enable the motor.
+        /// </summary>
+        /// <param name="Speed"> The new speed to set the motor at. From -1.0 to 1.0 </param>
+        public void SetSpeed(float Speed)
+        {
+            if (this.Filter != null && !this.Filter.IsSteadyState() && !OngoingSpeedThread)
+            {
+                this.Filter.Feed(Speed);
+                SetSpeedThreadFactory().Start();
+                OngoingSpeedThread = true;
+            }
+            else { SetSpeedDirectly(Speed); }
+            this.TargetSpeed = Speed;
+        }
+
+        /// <summary>
+        /// Sets the speed directly given an input from -1.0 to 1.0
+        /// Takes into consideration motor stop signal and max speed restriction.
+        /// </summary>
+        /// <param name="Speed"> Speed from -1.0 to 1.0 </param>
+        private void SetSpeedDirectly(float Speed)
+        {
+            if (Speed > this.MaxSpeed) { Speed = this.MaxSpeed; }
+            if (-Speed > this.MaxSpeed) { Speed = -this.MaxSpeed; }
+            if (this.Stopped) { Speed = 0; }
+            UARTBus.Write(ConstructPacket(Speed));
+        }
+
+
+        /// <summary>
+        /// Generates the packet for the motor controller:
+        /// One Start byte (value 2 for short packets and 3 for long packets)
+        /// One or two bytes specifying the packet length
+        /// The payload of the packet
+        /// Two bytes with a CRC checksum on the payload
+        /// One stop byte (value 3)
+        /// </summary>
+        /// <param name="Speed"> Speed from -1.0 to 1.0 </param>
+        private byte[] ConstructPacket(float Speed)
+        {
+            List<byte> packet = new List<byte>();
+            List<byte> payload = new List<byte>();
+            byte ShortPacket = 2;
+
+            packet.Add(ShortPacket); // Start byte

```

Comment:
No need to store `ShortPacket` in a `byte` if you're only going to use it once. You've integrated it together for the stop byte below, so just do that here too.

Is there an application where long packets may be needed? If so, handling this case should be implemented.

Reply:
Will fix first comment, to address second comment. No long packets will not be needed, the client will not be able to send custom messages over UART because the protocol is complicated.